llvm/lib/Target/Mips/MCTargetDesc/MipsTargetStreamer.cpp

```diff
@@ -1034,37 +1034,43 @@ MCELFStreamer &MipsTargetELFStreamer::getStreamer() {
 
 void MipsTargetELFStreamer::emitGPRel32Value(const MCExpr *Value) {
   auto &S = getStreamer();
+  S.ensureHeadroom(4);
```

Comment:
This looks like some kind of change that can easily cause hard-to-diagnose bugs when missed. This seems hard to catch with assertions, though, and I've no good idea how to improve this.

Reply:
I've replaced `MCFragment::appendContents` with `MCObjectStreamer::appendContents`, which calls `ensureHeadroom` internally. Therefore, the `ensureHeadroom` is redundant. I'll remove them.

 `MCFragment::appendContents`  has been removed. Hopefully it makes harder to make a mistake.